test/e2e/clusterlogging_test.go

```diff
@@ -279,3 +285,49 @@ func ClusterLoggingClusterRsyslog(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func clusterLoggingUpgradeClusterTest(t *testing.T, f *framework.Framework, ctx *framework.TestCtx, collector string) error {
+
+	namespace, err := ctx.GetNamespace()
+	if err != nil {
+		return fmt.Errorf("Could not get namespace: %v", err)
+	}
+
+	currentOperator, err := f.KubeClient.AppsV1().Deployments(namespace).Get("cluster-logging-operator", metav1.GetOptions{IncludeUninitialized: true})
+	if err != nil {
+		return fmt.Errorf("failed to get currentOperator: %v", err)
+	}
+
+	currentEnv := currentOperator.Spec.Template.Spec.Containers[0].Env
+	newEnv := []v1.EnvVar{
+		{Name: "WATCH_NAMESPACE", ValueFrom: &v1.EnvVarSource{FieldRef: &v1.ObjectFieldSelector{FieldPath: "metadata.namespace"}}},
+		{Name: "OPERATOR_NAME", Value: "cluster-logging-operator"},
+		{Name: "ELASTICSEARCH_IMAGE", Value: "docker.io/openshift/origin-logging-elasticsearch5:upgraded"},
+		{Name: "FLUENTD_IMAGE", Value: "docker.io/openshift/origin-logging-fluentd:upgraded"},
```

Comment:
This wont allow you to evaluate the need to upgrade the version.  Do we need a var to explicitly tell the operator to explicitly flush the buffers?

Reply:
This will tell it to upgrade the version, it doesn't cause it to do a buffer flush though. That may be outside the scope of this actually since that is part of the adoption work.
@richm thoughts?

Reply:
Will this upgrade the image from 3.11 to 4.0?

Reply:
We aren't advertising being able to upgrade from `3.11` to `4.0` currently, but yes, once we say we can do that if the current DS is at `3.11` based on the image name and the image provided on the `ENV` for the operator is `4.0` then it should trigger that it requires the buffer flush

Reply:
then I guess triggering the buffer flush it outside the scope of this work, and in a future PR we'll need to have some sort of logic like this:
- user wants to upgrade the image
- `oc set env ds/logging-fluentd FLUSH_AT_SHUTDOWN=true`
- restart fluentd on all nodes (the set env ds trigger usually restarts fluentd)
- shutdown fluentd on all nodes - will flush the buffers at shutdown
- change the image to the new version and set FLUSH_AT_SHUTDOWN=false
- start fluentd

Reply:
note that step `shutdown fluentd on all nodes - will flush the buffers at shutdown` may take a long, long time . . . depending on how much log buffer data there is

Reply:
So the way I have this written is that first we update the `ENV` to add `FLUSH_AT_SHUTDOWN=true` then we wait to have full availability again of the DS pods. After that we make the upgrade to `4.0` (this change would also remove the added `ENV` entry) which would cause another rollout, the operator doesn't wait on the pods after that.